src/NServiceBus.Core/Pipeline/Incoming/ITransportReceiveContext.cs

```diff
@@ -11,5 +12,11 @@ public interface ITransportReceiveContext : IBehaviorContext
         /// The physical message being processed.
         /// </summary>
         IncomingMessage Message { get; }
+    
+        /// <summary>
+        /// Allows the pipeline to flag that it has been aborted and the receive operation should be rolled back. 
+        /// It also allows the transport to communicate to the pipeline to abort if possible.
+        /// </summary>
+        CancellationTokenSource CancellationTokenSource { get; }
```

Comment:
@andreasohlund I kinda like the idea of `CancelReceiveOperation`, it seems to be easier to grasp and hides the implementation detail of using a cancellation token.

> No, we need to pipeline to be able to cancel so we need to pass the token source?

 the summary states:

> * Allows the pipeline to flag that it has been aborted and the receive operation should be rolled back. 

that's the cts, where the pipeline cancels the token source and it can be inspected by the transport.

> * It also allows the transport to communicate to the pipeline to abort if possible.

for that scenario the pipeline doesn't need to cancel a token source. It only needs to have a `CancellationToken` to check, right? That sounds like "missusing" a CTS for a "bidirectional" communication between transport and pipeline?